File: app_project/backend/app/schemas/news/scripture.py
from pydantic import BaseModel
from datetime import datetime

class ScriptureCategoryBase(BaseModel):
    name: str
    slug: str
    description: str | None = None

class ScriptureCategoryCreate(ScriptureCategoryBase):
    pass

class ScriptureCategoryUpdate(ScriptureCategoryBase):
    pass

class ScriptureCategoryResponse(ScriptureCategoryBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ScriptureBase(BaseModel):
    title: str
    author: str | None = None
    content: str | None = None
    pdf_url: str | None = None

class ScriptureCreate(ScriptureBase):
    category_id: int

class ScriptureUpdate(ScriptureBase):
    category_id: int | None = None

class ScriptureResponse(ScriptureBase):
    id: int
    category_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: app_project/frontend/src/models/news/scripture.ts
// ─── CATEGORY ─────────────────────────────────────────
export interface ScriptureCategoryBase {
  name: string;
  slug: string;
  description?: string | null;
}

export interface ScriptureCategoryCreate extends ScriptureCategoryBase {}
export interface ScriptureCategoryUpdate extends ScriptureCategoryBase {}

export interface ScriptureCategoryResponse extends ScriptureCategoryBase {
  id: number;
  created_at: string;
  updated_at: string;
}

// ─── SCRIPTURE ────────────────────────────────────────
export interface ScriptureBase {
  title: string;
  author?: string | null;
  content?: string | null;
  pdf_url?: string | null;
}

export interface ScriptureCreate extends ScriptureBase {
  category_id: number;
}

export interface ScriptureUpdate extends ScriptureBase {
  category_id?: number;
}

export interface ScriptureResponse extends ScriptureBase {
  id: number;
  category_id: number;
  created_at: string;
  updated_at: string;
}
